👁️ feat: Add Azure Mistral OCR strategy and endpoint integration

arthurolivierfortin opened the pull request:
This pull request introduces a new OCR strategy named azure_mistral_ocr, enabling support for Mistral OCR endpoints deployed on Azure. This allows seamless integration with Azure-hosted Mistral services, which currently do not support the default file upload API used by LibreChat.

To address this, the configuration, schema validations, and file handling logic have been updated to support this new strategy. It bypasses the default upload step and directly performs OCR through the Azure endpoint.

Fixes #6814


⚠️ Documentation Updates Notice:
The following configuration will need to be documented to enable users to adopt the new strategy:

```yaml
ocr:
  mistralModel: "mistral-ocr-2503"  # Optional: Specify Mistral model, defaults to "mistral-ocr-latest"
  apiKey: "${OCR_API_KEY}"        # Optional: Defaults to OCR_API_KEY env variable
  baseURL: "https://{azure_mistral_ocr_endpoint}/v1"
  strategy: azure_mistral_ocr
```

Once the PR is reviewed and approved, I’ll be happy to submit a separate PR to update the documentation accordingly.

## Summary

- Added a new OCR strategy: azure_mistral_ocr
- Added logic in api/server/services/Files/MistralOCR/crud.js to support Azure's behavior (bypassing pre-OCR file upload)
`api\server\services\Files\MistralOCR\crud.js`
- Extended schema and configuration validation to support the new strategy

## Change Type

Please delete any irrelevant options.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update


## Testing

- Set ocr.strategy to azure_mistral_ocr
- Use an mistral-ocr-2503 azure endpoint
- Upload a PDF or image using the Upload as Text option for an agent
- Send a query to the agent and verify the OCR output

### **Test Configuration**:

## Checklist

Please delete any irrelevant options.

- [x] My code adheres to this project's style guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented in any complex areas of my code
- [x] My changes do not introduce new warnings
- [x] Local unit tests pass with my changes
- [x] Any changes dependent on mine have been merged and published in downstream modules.
- [ ] ***A pull request for updating the documentation will submitted.

danny-avila commented:
Thanks @arthurolivierfortin the package.json changes should also not be included.

arthurolivierfortin committed:
👁️ feat: Add Azure Mistral OCR strategy and endpoint integration

This commit introduces a new OCR strategy named 'azure_mistral_ocr', allowing the use of a Mistral OCR endpoint deployed on Azure. The configuration, schemas, and file upload strategies have been updated to support this integration, enabling seamless OCR processing via Azure-hosted Mistral services.

arthurolivierfortin commented:
The modifications to package.json have been removed as requested.

danny-avila commented:
Reviewing this weekend, thanks for the work on this!

arthurolivierfortin commented:
Hi @danny-avila 
Just a quick follow-up in case this might have slipped through the cracks. No rush at all, if you need more time, that’s totally fine. I just wanted to make sure it hadn’t been forgotten.

Thanks again for taking the time to review!

arthurolivierfortin commented:
Totally understand, thanks for the update!

arthurolivierfortin commented:
Hi @danny-avila,

Thanks for adding the feature and for referencing my PR — I appreciate that my work could be helpful!

I just wanted to share a quick note regarding contributions.

When I open a PR with new features or improvements, it’s usually during my working hours. If the PR is closed without merging, or if the code is reimplemented without keeping my original commits, it becomes hard to justify that time internally — it can appear as paid work that didn’t make it into the project.

If possible, it would really help if in the future:
	•	the branch could be merged (or commits rebased) when the contribution is relevant, or
	•	there’s a clear indication of what parts will be kept or modified,
so I can adjust accordingly and align better with the project’s needs.

I genuinely enjoy contributing to LibreChat and would love to continue doing so. Making sure that contributions are acknowledged and tracked properly would really help my company support this involvement moving forward.

Thanks again, and feel free to let me know if you’d like to talk about it further!

danny-avila commented:
@arthurolivierfortin your original commit is in tact here: https://github.com/danny-avila/LibreChat/pull/7888/commits

I try to do this always to honor the original work, as you show up as a contributor still in the squashed commit. I only take over PRs to speed up the desired outcome, also because contributors may become unresponsive or not put in the effort of back and forth to complete. In this case, we had some back and forth already but wanted to expedite the completion.

arthurolivierfortin commented:
That would be great, thank you!

Just to be transparent, the features I’ve been adding are mostly based on internal needs from my company. That said, having collaborator access will definitely help me contribute more effectively moving forward and will likely increase my company’s interest in LibreChat as well.

I’m glad to be part of the project and excited to keep contributing.

danny-avila commented:
> @danny-avila Yeah, I was just about to ask if you’d be up for continuing this on Discord. Would you prefer I post in the suggestions channel or continue the discussion privately?

Nothing needs to be private, there is a #dev-talks channel and I also invite active contributors to a shared channel
